requirements.txt

```diff
@@ -14,3 +14,4 @@ autoray==0.3.1
 matplotlib~=3.5
 opt_einsum~=3.3
 requests~=2.28.1
```

Comment:
is it ok to upgrade this in this PR? there's a security vulnerability rn saying we need `requests>=2.31`. I haven't looked into it, but the data module is the only part of pennylane using it